reactive/containerd.py

```diff
@@ -458,7 +462,7 @@ def publish_version_to_juju():
     version_string = _check_containerd()
     if not version_string:
         return
-    version = version_string.split()[6].split(b"-")[0].decode()
+    version = version_string.split()[9].split(b"-")[0].decode()
```

Comment:
maybe it's just me but this seems magical 🦄 -- in a way that its indescribable.  

i kinda wish that `_check_containerd` was just called `_get_containerd_version()` and did all this splitting of lines and what not there.  

How do i know line 9 has the right text instead of line 6?  When does the line number change?  splitting on bytes?  There's a lot of interesting string work here

Reply:
Yes, renaming the `_check_containerd` method to `_get_containerd_version` and refactor the logic are great ideas!

`ctr version` output look like this:
```
Client:
Version:  1.5.9-0ubuntu1~20.04.4
Revision:
Go version: go1.13.8

Server:
Version:  1.5.9-0ubuntu1~20.04.4
Revision:
UUID: dc3fb3f1-3217-458b-8aaf-df2d7a4c7b91
```
as we can see, it outputs in the first 4 lines the Client information, then a newline that precedes the Server information. Therefore, containerd version will be always in the 9th line.

Regarding the bytes manipulation, `check_output` function returns bytes, so that's the reason of splitting on bytes. Nevertheless, I think changing to something like `subprocess.get_output()` could improve the code readability.

Reply:
What do you think about having something like:

```python
version_lines = set(line for line in output_lines if line.startswith("Version")])
```